File: web/src/features/campaigns/status.ts
/** Map a campaign's backend status to a StatusPill tone + label. */
export type CampaignTone = 'draft' | 'running' | 'paused' | 'done'

export function campaignTone(status?: string): CampaignTone {
  switch (status) {
    case 'running':
      return 'running'
    case 'paused':
      return 'paused'
    case 'done':
      return 'done'
    default:
      return 'draft'
  }
}

export function campaignLabel(status?: string): string {
  switch (status) {
    case 'running':
      return 'Running'
    case 'paused':
      return 'Paused'
    case 'done':
      return 'Done'
    default:
      return 'Draft'
  }
}

File: internal/app/campaign/status.go
// Package campaign manages campaign CRUD, ownership validation against
// mailboxes/lists, and (in a later task) launching sends.
package campaign

// CampaignStatus is the typed enum mirrored by the DB CHECK constraint on
// campaigns.status.
type CampaignStatus string

const (
	StatusDraft   CampaignStatus = "draft"
	StatusRunning CampaignStatus = "running"
	StatusPaused  CampaignStatus = "paused"
	StatusDone    CampaignStatus = "done"
)
